aiokafka/producer/message_accumulator.py

```diff
@@ -35,13 +36,27 @@ def append(self, *, timestamp, key, value):
         self._relative_offset += 1
         return actual_size
 
-    def _build(self):
-        assert not self._closed
+    def close(self):
+        if self._closed:
+            return
         self._closed = True
-        buffer = self._builder.build()
+        data = self._builder.build()
+        self._buffer = io.BytesIO(Int32.encode(len(data)) + data)
         del self._builder
 
-        return io.BytesIO(Int32.encode(len(buffer)) + buffer)
+    def _build(self):
+        if not self._closed:
+            self.close()
+        return self._buffer
+
+    def size(self):
+        if self._buffer:
+            return self._buffer.getbuffer().nbytes
+        else:
+            return self._builder.size()
+
+    def length(self):
```

Comment:
No, `length` and `size` have a similar meaning and it's confusing. How about `record_count()`.